src/pages/workerDetails/WorkerDetails.jsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from "react";
+import { Link, useParams } from "react-router-dom";
+
+export const WorkerDetails = () => {
+  const { id } = useParams();
+  const [workerInfoDetails, setWorkerInfoDetails] = useState([]);
```

Comment:
If you'd like to have a loading scenario, consider useState(null), where null value represents no data. You will also have to add additional logic, not to attempt data maping, until null chas changed into an object.